packages/kbn-text-based-editor/src/text_based_languages_editor.tsx

```diff
@@ -185,18 +196,43 @@ export const TextBasedLanguagesEditor = memo(function TextBasedLanguagesEditor({
     isCodeEditorExpanded ? EDITOR_INITIAL_HEIGHT_EXPANDED : EDITOR_INITIAL_HEIGHT
   );
   const [isSpaceReduced, setIsSpaceReduced] = useState(false);
+  const [editorWidth, setEditorWidth] = useState(0);
+  const [isHistoryOpen, setIsHistoryOpen] = useState(false);
   const [showLineNumbers, setShowLineNumbers] = useState(isCodeEditorExpanded);
   const [isCompactFocused, setIsCompactFocused] = useState(isCodeEditorExpanded);
   const [isCodeEditorExpandedFocused, setIsCodeEditorExpandedFocused] = useState(false);
   const [isQueryLoading, setIsQueryLoading] = useState(true);
   const [abortController, setAbortController] = useState(new AbortController());
+  // contains both client side validation and server messages
   const [editorMessages, setEditorMessages] = useState<{
     errors: MonacoMessage[];
     warnings: MonacoMessage[];
   }>({
     errors: serverErrors ? parseErrors(serverErrors, code) : [],
     warnings: serverWarning ? parseWarning(serverWarning) : [],
   });
+  // contains only client side validation messages
+  const [clientParserMessages, setClientParserMessages] = useState<{
```

Comment:
The maps component is propagating as an error when a query doesnt have a geo field. The query is valid but it is not "valid" for maps. The client side validator though knows if a query is valid or not. So for the history component status I am going to use these messages instead of the errors that each application passes to the errors / warning props as in some cases is the most correct source of truth.

Reply:
Makes sense. Context complicates things 👍